workers/social/lib/social/models/group/index.coffee

```diff
@@ -1493,65 +1493,106 @@ module.exports = class JGroup extends Module
+          removeHelper = (model, err, next) ->
+            return next err  if err
+            return next()  unless model
 
-        (next) =>
-          @fetchPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+            model.remove (err) -> next err
 
-        (next) =>
-          @fetchDefaultPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+          removeHelperMany = (klass, models, err, next) ->
+            return next err  if err
+            return next()    if not models or models.length < 1
 
-        (next) =>
-          JInvitation = require '../invitation'
-          JInvitation.remove { groupName: @slug }, (err) ->
-            next err
+            ids = (model._id for model in models)
+            klass.remove ({ _id: { $in: ids } }), (err) -> next err
 
-        (next) =>
-          ComputeProvider = require '../computeproviders/computeprovider'
-          ComputeProvider.destroyGroupResources this, -> next()
+          async.series [
 
-        (next) =>
-          JSession = require '../session'
-          @sendNotification 'GroupDestroyed', @slug, =>
-            JSession.remove { groupName: @slug }, (err) ->
-              next err
+            (next) ->
+              JName.one { name: slug }, (err, name) ->
+                removeHelper name, err, next
 
-        (next) =>
-          @constructor.emit 'GroupDestroyed', this
-          next()
+            (next) =>
+              @fetchPermissionSet (err, permSet) ->
+                removeHelper permSet, err, next
+
+            (next) =>
+              @fetchDefaultPermissionSet (err, permSet) ->
+                removeHelper permSet, err, next
+
+            (next) ->
+              JInvitation = require '../invitation'
+              JInvitation.remove { groupName: slug }, (err) ->
+                next err
+
+            (next) =>
+              ComputeProvider = require '../computeproviders/computeprovider'
+              ComputeProvider.destroyGroupResources this, -> next()
+
+            (next) ->
+              url = '/api/social/payment/subscription/delete'
+              { deleteReq } = require '../socialapi/requests'
+
+              deleteReq url, { sessionToken }, (err, body) ->
+                next err
+
+            (next) =>
+              JSession = require '../session'
+              @sendNotification 'GroupDestroyed', slug, ->
+                JSession.remove { groupName: slug }, (err) ->
+                  next err
+
+            (next) ->
+              JApiToken = require '../apitoken'
+              JApiToken.remove { group: slug }, (err) ->
+                next err
+
+            (next) =>
+              @constructor.emit 'GroupDestroyed', this
+              next()
 
-        (next) =>
-          @remove (err) -> next err
+            (next) =>
+              @remove (err) -> next err
 
-      ], callback
+          ], (err) =>
+            @sendNotification 'GroupDestroyed'  if err
```

Comment:
If any error is occured before this line https://github.com/koding/koding/pull/10521/files#diff-23aa32e384f44c80809d4a6cb4c27095R1577, we will delete user's cookies and redirect to landing page.